Compilation failure in libsinsp/filter/ast.cpp with gcc-15: uint32_t does not name a type

**Describe the bug**

In Gentoo we're test-driving early gcc-15 releases to shake out bugs. Our QA has found an issue when building libs-0.17.3 as part of sysdig: [dev-debug/sysdig-0.38.1-r1 - [gcc-15] /.../ast.h: error: const struct libsinsp::filter::ast::pos_info has no member named idx](https://bugs.gentoo.org/938218)

**How to reproduce it**

Try to build libs with gcc-15, it should error out with:
```
FAILED: libsinsp/CMakeFiles/sinsp.dir/filter/ast.cpp.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DCHISEL_TOOL_LIBRARY_NAME=\"sysdig\" -DHAS_CAPTURE -DHAVE_GRP_H -DHAVE_PWD_H -DHAVE_STRLCPY -DPLATFORM_NAME=\"Linux\" -DSINSP_AGENT_CGROUP_MEM_PATH_ENV_VAR=\"AGENT_CGROUP_MEM_PATH\" -D_GNU_SOURCE -I/usr/include/tbb -I/usr/include/jsoncpp -I/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp -I/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace -I/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libscap -I/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/sysdig-0.38.1_build/libscap -I/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3 -I/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/sysdig-0.38.1_build -I/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libscap/linux -I/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/sysdig-0.38.1_build/libsinsp  -O2 -pipe -march=native -fno-diagnostics-color -fno-strict-aliasing -Wall    -std=c++17 -std=c++17 -fno-diagnostics-color -MD -MT libsinsp/CMakeFiles/sinsp.dir/filter/ast.cpp.o -MF libsinsp/CMakeFiles/sinsp.dir/filter/ast.cpp.o.d -o libsinsp/CMakeFiles/sinsp.dir/filter/ast.cpp.o -c /var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.cpp
In file included from /var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.cpp:19:
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:52:22: error: expected ‘)’ before ‘i’
   52 |     pos_info(uint32_t i, uint32_t l, uint32_t c): idx(i), line(l), col(c) { }
      |             ~        ^~
      |                      )
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:80:5: error: ‘uint32_t’ does not name a type
   80 |     uint32_t idx = 0;
      |     ^~~~~~~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:26:1: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
   25 | #include <libsinsp/sinsp_public.h>
  +++ |+#include <cstdint>
   26 | 
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:81:5: error: ‘uint32_t’ does not name a type
   81 |     uint32_t line = 1;
      |     ^~~~~~~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:81:5: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:82:5: error: ‘uint32_t’ does not name a type
   82 |     uint32_t col = 1;
      |     ^~~~~~~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:82:5: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h: In member function ‘bool libsinsp::filter::ast::pos_info::operator==(const libsinsp::filter::ast::pos_info&) const’:
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:59:16: error: ‘idx’ was not declared in this scope
   59 |         return idx == b.idx && line == b.line && col == b.col;
      |                ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:59:25: error: ‘const struct libsinsp::filter::ast::pos_info’ has no member named ‘idx’
   59 |         return idx == b.idx && line == b.line && col == b.col;
      |                         ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:59:32: error: ‘line’ was not declared in this scope
   59 |         return idx == b.idx && line == b.line && col == b.col;
      |                                ^~~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:59:42: error: ‘const struct libsinsp::filter::ast::pos_info’ has no member named ‘line’
   59 |         return idx == b.idx && line == b.line && col == b.col;
      |                                          ^~~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:59:50: error: ‘col’ was not declared in this scope
   59 |         return idx == b.idx && line == b.line && col == b.col;
      |                                                  ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:59:59: error: ‘const struct libsinsp::filter::ast::pos_info’ has no member named ‘col’
   59 |         return idx == b.idx && line == b.line && col == b.col;
      |                                                           ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h: In member function ‘bool libsinsp::filter::ast::pos_info::operator!=(const libsinsp::filter::ast::pos_info&) const’:
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:63:16: error: ‘idx’ was not declared in this scope
   63 |         return idx != b.idx || line != b.line || col != b.col;
      |                ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:63:25: error: ‘const struct libsinsp::filter::ast::pos_info’ has no member named ‘idx’
   63 |         return idx != b.idx || line != b.line || col != b.col;
      |                         ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:63:32: error: ‘line’ was not declared in this scope
   63 |         return idx != b.idx || line != b.line || col != b.col;
      |                                ^~~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:63:42: error: ‘const struct libsinsp::filter::ast::pos_info’ has no member named ‘line’
   63 |         return idx != b.idx || line != b.line || col != b.col;
      |                                          ^~~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:63:50: error: ‘col’ was not declared in this scope
   63 |         return idx != b.idx || line != b.line || col != b.col;
      |                                                  ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:63:59: error: ‘const struct libsinsp::filter::ast::pos_info’ has no member named ‘col’
   63 |         return idx != b.idx || line != b.line || col != b.col;
      |                                                           ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h: In member function ‘void libsinsp::filter::ast::pos_info::reset()’:
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:68:9: error: ‘idx’ was not declared in this scope
   68 |         idx = 0;
      |         ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:69:9: error: ‘line’ was not declared in this scope
   69 |         line = 1;
      |         ^~~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:70:9: error: ‘col’ was not declared in this scope
   70 |         col = 1;
      |         ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h: In member function ‘std::string libsinsp::filter::ast::pos_info::as_string() const’:
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:75:42: error: ‘idx’ was not declared in this scope
   75 |         return "index " + std::to_string(idx)
      |                                          ^~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:76:42: error: ‘line’ was not declared in this scope
   76 |             + ", line " + std::to_string(line)
      |                                          ^~~~
/var/tmp/portage/dev-debug/sysdig-0.38.1-r1/work/libs-0.17.3/userspace/libsinsp/filter/ast.h:77:44: error: ‘col’ was not declared in this scope
   77 |             + ", column " + std::to_string(col);
      |                                            ^~~
```

**Expected behaviour**

Successful build :rocket: 

**Environment**

- OS:

Any Gentoo Linux with gcc-15

**Additional context**

gcc-15's libstdc++ has become more strict and no longer automagically pulls in `<cstdint>`, see [here](https://gcc.gnu.org/pipermail/gcc-patches/2024-August/659176.html) for the gcc mailing list thread.

The error cascade looks scarier than it is, since - as the compiler explains - the root of the problem is just a missing `#include <cstdint>` in `libsinsp/filter/ast.h`.

--- a/userspace/libsinsp/filter/ast.h
+++ b/userspace/libsinsp/filter/ast.h
@@ -18,6 +18,7 @@ limitations under the License.
 
 #pragma once
 
+#include <cstdint>
 #include <vector>
 #include <string>
 #include <algorithm>

